common/OpTestUtil.py

```diff
@@ -1913,6 +1913,112 @@ def wait_for(self, func, timeout, first=0.0, step=1.0, text=None, args=None, kwa
             time.sleep(step)
         return None
 
+    def clear_dmesg(self):
+        host = self.conf.host()
+        host.host_run_command("dmesg -C")
+
+    def collect_errors_by_level(self, output_file=None, level_check=5, skip_errors=None):
+        """
+        Verify dmesg having severity level of OS issue(s).
+
+        :param output_file: The file used to save dmesg
+        :type output_file: str
+        :param level_check: level of severity of issues to be checked
+                            1 - emerg
+                            2 - emerg,alert
+                            3 - emerg,alert,crit
+                            4 - emerg,alert,crit,err
+                            5 - emerg,alert,crit,err,warn
+        :type level_check: int
+        :skip_errors: list of dmesg error messages which want skip
+        :type skip_errors: list
+        """
+        host = self.conf.host()
+        dmsg_log = out = ""
+        cmd = "dmesg -T -l %s|grep ." % ",".join(map(str, range(0, int(level_check))))
+        try:
+            out = '\n'.join(host.host_run_command(cmd, timeout=30))
+        except CommandFailed as cmd_failed:
+            if cmd_failed.exitcode == 1 and len(cmd_failed.output) == 0:
+                pass
+            else:
+                raise cmd_failed
+        dmsg_log = self.skip_dmesg_messages(out, skip_errors) if skip_errors else out
+        if dmsg_log:
+            if output_file:
+                output_dir = os.path.join(host.results_dir, "DmesgErrorLogs")
+                if not os.path.exists(output_dir):
+                    os.makedirs(output_dir)
+                output_file = os.path.join(output_dir, output_file)
+                with open(output_file, "w+", encoding='utf-8') as log_f:
+                    log_f.write(dmsg_log)
+                err = "Found failures in dmesg. Please check dmesg log %s." % (output_file)
+            else:
+                err = "Found failures in dmesg. Please check dmesg log in debug log."
```

Comment:
Can be changed to "Found failures in dmesg. Please check debug log."

Reply:
Done